enhancements/network/smart-nic-ovn-offload.md

```diff
@@ -0,0 +1,389 @@
+This is currently done through ssh or console from SmartNIC node using rshim
+[userspace driver](https://bugzilla.redhat.com/show_bug.cgi?id=1744737).
+Rshim userspace driver needs to be made availabe in RHCOS image or downloadable
+in Universal Base Image (UBI).
+
+#### OpenShift and RHEL support for aarch64
+
+Management of SmartNIC ARM cores (aarch64) as individual node in a separate
+OpenShift cluster requires OpenShift and RHEL support for aarch64.
+
+#### Manage SmartNICs and SmartNIC nodes in separate clusters
+
+This enhancement makes the assumption that both SmartNICs and SmartNIC nodes
+belong to the same OpenShift cluster, where SmartNIC nodes are managed by
+OpenShift and SmartNICs are managed by user. Any future changes that beak this
+assumption will need an update in the enhancement.
+
+## Design Details
+
+### Open Questions
+
+With current OVN-Kubernetes design, it is only supported to run SR-IOV pods on
+the SmartNIC node. Regular pods whose networks require virtual **veth** interfaces
+cannot be plugged to OVN network because there will be no Open vSwitch running
+on the SmartNIC node.
+
+To create a SR-IOV pod, it is needed to request the SR-IOV resource explicitly
+in pod spec, for example:
+
+```yaml
+apiVersion: v1
+kind: Pod
+metadata:
+  name: sriov-pod
+spec:
+  containers:
+  - name: sriov-container
+    image: centos
+    resources:
+      requests:
+        openshift.io/<sriov-resource-name>: 1
+      limits:
+        openshift.io/<sriov-resource-name>: 1
+```
+
+In the above pod yaml file:
+
+`openshift.io/<sriov-resource-name>` is the full resource name that follows
+the [extended resource naming scheme](https://kubernetes.io/docs/concepts/configuration/manage-resources-containers/#extended-resources),
+it is advertised by SR-IOV Network Device Plugin and published in Kubernetes
+node status. The SR-IOV Network Device Plugin is a component that discovers
+and manages the limited SR-IOV devices on the SmartNIC node. The managed
+SR-IOV devices, AKA VFs, are created from SmartNIC PF.
+
+`1` is the number of SR-IOV devices requested for the `sriov-pod`. When the
+requested number of SR-IOV devices are not available on the target node, the
+`sriov-pod` will be in `Pending` state until resource become available again
+(e.g. other SR-IOV pods release the VF device on the target node).
+
+Given that regular **veth** pods cannot be created on the SmartNIC node,
+failure would happen if some cluster pods or infra pods using **veth**
+interfaces are scheduled on the SmartNIC node, examples of cluster or infra
+pods are as below:
+
+- ingress-canary in openshift-ingress-canary namespace
+- image-pruner in openshift-image-registry namespace
+- network-metrics-daemon in openshift-multus namespace
+- network-check-source & network-check-target in openshift-network-diagnostics
+- Pods in openshift-marketplace & openshift-monitoring namespaces
+
+There are several possible ways to avoid the failure:
+
```

Comment:
Another possible option could be tainting the smart-nic nodes, so that regular pods without the corresponding toleration won't be scheduled to those nodes.

Reply:
@pliurh I think this is covered in the second bullet, wdyt? 
We could use taints, labels or what other selectors to select the node where cluster pods will be scheduled on.

Reply:
Hmmm, in the second bullet, it says 'modify cluster pod manifests. However, if we taint the node, we don't need to modify the pod manifests.

Reply:
You're right, updated.